src/journal/journalctl.c

```diff
@@ -2079,6 +2080,64 @@ typedef struct Context {
         dual_timestamp previous_ts_output;
 } Context;
 
+static int verify_realtime(Context *c) {
+        sd_journal *j = ASSERT_PTR(ASSERT_PTR(c)->journal);
+        usec_t usec;
+        int r;
+
+        /* This returns -EDOM if the caller should exit from the loop, 0 if the caller should skip the
+         * current journal entry, 1 if the check passes. */
+
+        /* Both --since and --until are unspecified. It is not necessary to check the timestamp. */
+        if (!arg_until_set && !arg_since_set)
+                return 1;
+
+        /* Only --until is specified, and we know that the current position is in the range. */
+        if (arg_until_set && !arg_since_set && c->until_seeked && arg_reverse)
+                return 1;
+
+        /* Only --since is specified, and we know that the current position is in the range. */
+        if (!arg_until_set && arg_since_set && c->since_seeked && !arg_reverse)
+                return 1;
+
+        r = sd_journal_get_realtime_usec(j, &usec);
+        if (r < 0)
+                return log_error_errno(r, "Failed to determine realtime timestamp: %m");
+
+        if (arg_until_set && usec > arg_until) {
+                if (c->until_seeked)
```

Comment:
So this variable is kinda redundant and unnecessary? Essentially this can still be derived from arg_reverse. With this condition, things actually get more complex: if the current seek is after arg_until, we actually first seek to arg_until, and in the next loop return -EDOM?

Reply:
Maybe, the new verifier is redundant or less optimal. But the current logic is already too complicated, and I'd like to make that simpler and easily understood.
And the possibility you mentioned is not performance sensitive, I think.
At least, the logic by this PR should be much simpler and symmetric, this hopefully easily understood.